utils/frame/benchmarking-cli/src/writer.rs

```diff
@@ -145,27 +147,45 @@ fn get_benchmark_data(batch: &BenchmarkBatch) -> BenchmarkData {
 	let mut used_reads = Vec::new();
 	let mut used_writes = Vec::new();
 
-	extrinsic_time.slopes.into_iter().zip(extrinsic_time.names.iter()).for_each(|(slope, name)| {
-		if !slope.is_zero() {
-			if !used_components.contains(&name) { used_components.push(name); }
-			used_extrinsic_time.push(ComponentSlope {
-				name: name.clone(),
-				slope: slope.saturating_mul(1000),
-			});
-		}
-	});
-	reads.slopes.into_iter().zip(reads.names.iter()).for_each(|(slope, name)| {
-		if !slope.is_zero() {
-			if !used_components.contains(&name) { used_components.push(name); }
-			used_reads.push(ComponentSlope { name: name.clone(), slope });
-		}
-	});
-	writes.slopes.into_iter().zip(writes.names.iter()).for_each(|(slope, name)| {
-		if !slope.is_zero() {
-			if !used_components.contains(&name) { used_components.push(name); }
-			used_writes.push(ComponentSlope { name: name.clone(), slope });
-		}
-	});
+	extrinsic_time.slopes.into_iter()
+		.zip(extrinsic_time.names.iter())
+		.zip(extrinsic_time.model.unwrap().se.regressor_values.iter())
+		.for_each(|((slope, name), error)| {
+			if !slope.is_zero() {
```

Comment:
Ahh fuck :P  This could have been a `filter()`. But yeah, not that important xD

Reply:
Would it actually be better given that i need to use all the variables in the code block below?

I would need a `filter` statement on `slope` and a `then` statement looking at all the variables again?

Reply:
You would just remove the if. Nothing else would change.